Merge Mip 69

rdenarie opened the pull request.

hakermi committed:
feat: Exclude bcprov-jdk15on from wallet packaging - EXO-59676 - Meeds-io/MIPs#69  (#395)

Prior to this change, bcprov-jdk15on jar is added in wallet packaging as
it's coming as a compiled dependency in org.web3j.utils.

With new feature about the new Argon2IdPasswordEncoder, this dependency
is needed at gatein portal level. So this PR exclude this jar from
compiled scope, so that, if someone uninstall wallet addon, the
bcprov-jdk15on is not removed.

boubaker commented on wallet-packaging/src/main/assemblies/assembly.xml:
```diff
@@ -40,6 +40,9 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
       <includes>
         <include>*:*:jar</include>
       </includes>
+      <excludes>
+        <exclude>org.bouncycastle:bcprov-jdk15on:jar</exclude>
+      </excludes>
```
```suggestion
```
Add this exclusion in POM instead of here to exclude its transitive dependency as well, else only the jar will be excluded and not its transitive dependencies.

rdenarie replied:
Done

rdenarie committed:
feat: Exclude bcprov-jdk15on from wallet packaging - EXO-59676 - Meeds-io/MIPs#69  (#395)

Prior to this change, bcprov-jdk15on jar is added in wallet packaging as
it's coming as a compiled dependency in org.web3j.utils.

With new feature about the new Argon2IdPasswordEncoder, this dependency
is needed at gatein portal level. So this PR exclude this jar from
compiled scope, so that, if someone uninstall wallet addon, the
bcprov-jdk15on is not removed.

This commit move the exclude in the pom in order to also exlude transitive dependencies
